zebra/zebra_nhg.c

```diff
@@ -221,6 +221,17 @@ static int nexthop_active(afi_t afi, struct route_entry *re,
 		}
 	}
 
```

Comment:
Could there be a comment about this specific case? looking at the whole function, this block and the one later in the loop look quite a bit similar - and that one's comment seems to apply to exactly the case we're trying to guard against in the new addition.

Reply:
That one is slightly different. In it, we are trying to recursively resolve to ourselves. This is a route that explicitly points its gateway to the route its trying to add.

Reply:
But I agree, a comment is owed here.